### DISCUSSION TITLE
Buttons with Command and CommandParameter no longer work with v11?

### DISCUSSION THREAD

**Discussion by sn4k3** [AvaloniaUI/Avalonia#12062] [Q&A] [answered]
While upgrading my app to v11 I found that buttons/menuitems with `Command` no longer works as long `CommandParameter` is set. I Didn't notice any breaking change regarding this usage.

For example, in image below all greyed-out items are buttons with a `Command` with a `CommandParameter`.
Buttons that are not disabled just have `Command` without a `CommandParameter`.

![image](https://github.com/AvaloniaUI/Avalonia/assets/113281/9b330316-84c0-4c44-bf7f-114163a4d6cd)

Is there a solution for that?

**Comment by maxkatz6** [comment, marked as answer]
Please double check this part:

 https://docs.avaloniaui.net/docs/next/stay-up-to-date/upgrade-from-0.10

> When binding to methods, the method must either have no parameters or a single object parameter.

**Reply by sn4k3** [reply]
Worked, thanks!

**Reply by aerologos** [reply]
> Please double check this part:
> 
> https://docs.avaloniaui.net/docs/next/stay-up-to-date/upgrade-from-0.10
> 
> > When binding to methods, the method must either have no parameters or a single object parameter.

I love you man.

**Reply by sn4k3** [reply]
I also discovered that you can use a typed argument and bind to command, but you need to pass the type

```xml
<controls:ButtonWithIcon.CommandParameter>
    <system:Int32>5000</system:Int32>
</controls:ButtonWithIcon.CommandParameter>
```

**Reply by maxkatz6** [reply]
Makes sense, otherwise it would be treated as string.

**Comment by Shivansps** [comment]
I had the problem were ALL my buttons where disabled, thats because i had set all my command methods to private, witch in v10 works.
